profiler/profile.go

```diff
@@ -43,50 +44,143 @@ const (
 	expGoroutineWaitProfile
 	// MetricsProfile reports top-line metrics associated with user-specified profiles
 	MetricsProfile
+	// Adding new profiles also needs to happen in the profileTypes map and
+	// profiler.enabledProfileTypes().
```

Comment:
This is not so great. Is there any way we can help prevent potential programming errors by using different structures or tests?

If someone adds a constant somewhere in the middle of this list, there's a very good chance they might miss this comment. Perhaps the map could be checked by running through all constants suffixed `*Profile` using `go/ast`. Is that overkill? Do you think it's worth it? I'm not sure off the top of my head how much code that'd be but probably not that complicated.

Reply:
Yeah, it's not ideal, but I think it's ok. Any new profile type should be covered by a test case. Failure to follow the recommendations made in this comment will prevent the test case from passing.

Reply:
See 3fcf217